# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/WebrtcTCPSocketWrapper.cpp
# Commit: c43589fd0f90
# Full Hash: c43589fd0f90d174eb185b563406fd0a2e6ea712
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Update OnConnected callbacks to specify what type of proxying (if any) is being used, and implement IsProxied based on this information. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45186
# ==============================================================================

diff -r 940d7f787bea -r c43589fd0f90 media/mtransport/WebrtcTCPSocketWrapper.cpp
--- a/media/mtransport/WebrtcTCPSocketWrapper.cpp	Tue Sep 17 18:18:15 2019 +0000
+++ b/media/mtransport/WebrtcTCPSocketWrapper.cpp	Tue Sep 17 18:18:33 2019 +0000
@@ -125,13 +125,13 @@
           &WebrtcTCPSocketCallback::OnRead, std::move(aReadData))));
 }
 
-void WebrtcTCPSocketWrapper::OnConnected() {
+void WebrtcTCPSocketWrapper::OnConnected(const nsCString& aProxyType) {
   MOZ_ASSERT(NS_IsMainThread(), "not on main thread");
   MOZ_ASSERT(mProxyCallbacks, "webrtc TCP callbacks should be non-null");
 
-  MOZ_ALWAYS_SUCCEEDS(mSocketThread->Dispatch(
-      NewRunnableMethod("WebrtcTCPSocketWrapper::OnConnected", mProxyCallbacks,
-                        &WebrtcTCPSocketCallback::OnConnected)));
+  MOZ_ALWAYS_SUCCEEDS(mSocketThread->Dispatch(NewRunnableMethod<nsCString>(
+      "WebrtcTCPSocketWrapper::OnConnected", mProxyCallbacks,
+      &WebrtcTCPSocketCallback::OnConnected, aProxyType)));
 }
 
 }  // namespace net